packages/apps/human-app/frontend/src/modules/worker/components/profile/components/wallet-section.tsx

```diff
@@ -0,0 +1,51 @@
+import { useTranslation } from 'react-i18next';
+import { Button } from '@/shared/components/ui/button';
+import { DoneLabel } from '@/modules/worker/components/profile/components/status-labels/done-label';
+import { WalletConnectDone } from '@/modules/worker/components/profile/components/wallet-connect-done';
+import type { ProfileStatus } from '@/modules/worker/components/profile/types/profile-types';
+
+interface WalletSectionProps {
+  status: Pick<ProfileStatus, 'kycApproved'>;
+  isConnected: boolean;
+  hasWalletAddress: boolean;
+  isRegisterAddressPending: boolean;
+  onConnect: () => void;
+}
+
+export function WalletSection({
+  status,
+  isConnected,
+  hasWalletAddress,
+  isRegisterAddressPending,
+  onConnect,
+}: WalletSectionProps) {
+  const { t } = useTranslation();
+
+  if (!status.kycApproved) {
+    return (
+      <Button disabled fullWidth variant="contained">
+        {t('components.wallet.connectBtn.connect')}
+      </Button>
+    );
+  }
+
+  if ((!hasWalletAddress && isConnected) || hasWalletAddress) {
```

Comment:
This can be simplified:

```suggestion
  if (isConnected || hasWalletAddress) {
```

Reply:
good point :D